esp/services/ws_fs/ws_fsBinding.cpp

```diff
@@ -370,6 +300,54 @@ IPropertyTree* CFileSpraySoapBindingEx::createPTreeForXslt(const char* method, c
     return pRoot.getClear();
 }
 
+//For every ECLWatchVisible dropzones, read: dropZoneName, directory, and, if available, computer.
+//For every Servers/Server in ECLWatchVisible dropzones, read: directory,
+// server name, and hostname(or IP). Create dropzone("@name", "@directory", "@computer",
+// "@netAddress", "@linux", "@sourceNode") tree into pSoftware.
+void CFileSpraySoapBindingEx::appendDropZones(double clientVersion, IConstEnvironment* env, const char* dfuwuidSourcePartIP, IPropertyTree* softwareTree)
+{
+    IArrayOf<IConstTpDropZone> list;
+    CTpWrapper dummy;
+    dummy.getTpDropZones(clientVersion, nullptr, true, list);
```

Comment:
Ok, I saw that the getDropZone() in the environment interfaces has already handled the legacy dropzone/machine format. Here is one functionality it does not have: the existing ws_fs code assumes that the Directory setting for a dropzone may not use correct path separator ('/' vs '\\'). The ws_fs code will correct it based on computer name -> OS.  Should we add this functionality to  the environment interfaces?